packages/web-components/src/components/filter-panel/filter-panel-composite.ts

```diff
@@ -412,25 +426,32 @@ class DDSFilterPanelComposite extends HostListenerMixin(
    */
   protected _renderDesktop = (): TemplateResult => html`
     <dds-filter-panel
-      heading="${this._filterButtonTitle}"
+      heading="${this._getComposedHeadingFilterCount()}"
       ?has-selections="${this._selectedValues.length}">
-      ${this._title.map((e) => {
-        return html` ${unsafeHTML((e as HTMLElement).outerHTML)} `;
-      })}
-      ${this._contents.map((e) => {
-        return html` ${unsafeHTML((e as HTMLElement).outerHTML)} `;
-      })}
+      <slot @slotchange="${this._handleSlotChange}"></slot>
     </dds-filter-panel>
   `;
 
   render() {
+    // Note that the <slot name="heading"> contents, intended to be
+    // <dds-filter-panel-heading> are never shown as is. The text contents
+    // are composed, using this._getComposedHeadingFilterCount(), together with
+    // the current filter count, and passed as an attribute to
+    // <dds-filter-panel-modal> and <dds-filter-panel>.
```

Comment:
Thinking out loud, are we set up to essentially deprecate the heading slot altogether? We're doing a lot of work to simply harvest the text content of the first element in this slot, and perhaps this is a scenario where it makes more sense to have a `heading` attribute. Users set it, it gets reflected into a property we access to compose the heading + filter count.

Reply:
Totally agree. I only kept the slot b/c getting rid of it would break compatibility. Are you suggesting that we build in support for a `heading` attribute here and mark the slot and `<dds-filter-panel-heading>` as deprecated? I could get behind that.

Reply:
Yes, exactly! Let's do it 💪

Reply:
Cool, I'll file a separate issue / PR pair for that to follow this up. Feels worth the paper trail.

Reply:
Filed: https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/11062